app/src/main/java/org/sopt/sample/presentation/login/SignInViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package org.sopt.sample.presentation.login
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import org.sopt.sample.entity.User
+import org.sopt.sample.util.SaveUserInfo
+import javax.inject.Inject
+
+@HiltViewModel
+class SignInViewModel @Inject constructor(
+    private val saveUserInfo: SaveUserInfo
+) : ViewModel() {
+    private val _userInfo = MutableLiveData<User>()
+    val userInfo: LiveData<User> = _userInfo
+
+    val inputId = MutableLiveData<String>()
+    val inputPwd = MutableLiveData<String>()
+    val isAutoLogin = MutableLiveData<Boolean>()
+
+    private val _successLogin = MutableLiveData<Boolean>()
+    val successLogin: LiveData<Boolean> = _successLogin
+
+    init {
+        viewModelScope.launch {
+            if (saveUserInfo.getAutoLogin()) {
+                _userInfo.value = saveUserInfo.getUserInfo()
+                _successLogin.value = true
+            }
+            Log.d(
+                "asdf",
+                "SignInViewModel, 자동로그인:${saveUserInfo.getAutoLogin()}, 유저정보:${saveUserInfo.getUserInfo()}"
+            )
+        }
+    }
+
+    fun loginOnClick() {
+        viewModelScope.launch {
+            if (inputId.value == userInfo.value?.id && inputPwd.value == userInfo.value?.pwd) {
+                if (isAutoLogin.value!!) {
```

Comment:
```suggestion
                if (isAutoLogin.value ?: false) {
```

null 때문에 터질 수 있을 것 같아

Reply:
> null 때문에 터질 수 있을 것 같아

헉 그르네요 감사함다